### PR TITLE
Add support for Razer BlackWidow V3 Pro (Wireless)

### LINKED ISSUE #1279: Support for BlackWidow V3 Pro [1532:025c]
### Device Info

* Name: BlackWidow V3 Pro
* Product Number: RZ03-03530200-R3U1
* Model Number: RZ03-0353

### Device Information

<details>
  <summary> lsusb -d 1532: -v </summary>

  ```
Bus 001 Device 003: ID 1532:025c Razer USA, Ltd Razer BlackWidow V3 Pro
Couldn't open device, some information will be missing
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x025c 
  bcdDevice            1.00
  iManufacturer           1 
  iProduct                2 
  iSerial                 3 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0086
    bNumInterfaces          5
    bConfigurationValue     1
    iConfiguration          4 
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     179
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      94
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        4
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     325
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x85  EP 5 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        3
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     348
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x84  EP 4 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
  ```
</details>
    
<details>
  <summary>ls -lah /dev/input/by-id/</summary>

  ```
total 0
drwxr-xr-x 2 root root 240 Oct  3 20:34 .
drwxr-xr-x 4 root root 800 Oct  3 20:34 ..
lrwxrwxrwx 1 root root   9 Oct  1 17:28 usb-Logitech_USB_Receiver-if02-event-kbd -> ../event6
lrwxrwxrwx 1 root root   9 Oct  1 17:28 usb-Logitech_USB_Receiver-if02-event-mouse -> ../event4
lrwxrwxrwx 1 root root   9 Oct  1 17:28 usb-Logitech_USB_Receiver-if02-mouse -> ../mouse2
lrwxrwxrwx 1 root root  10 Oct  3 20:34 usb-Razer_Razer_BlackWidow_V3_Pro_000000000000-event-if01 -> ../event27
lrwxrwxrwx 1 root root  10 Oct  3 20:34 usb-Razer_Razer_BlackWidow_V3_Pro_000000000000-event-if03 -> ../event29
lrwxrwxrwx 1 root root  10 Oct  3 20:34 usb-Razer_Razer_BlackWidow_V3_Pro_000000000000-event-kbd -> ../event23
lrwxrwxrwx 1 root root  10 Oct  3 20:34 usb-Razer_Razer_BlackWidow_V3_Pro_000000000000-if01-event-kbd -> ../event24
lrwxrwxrwx 1 root root  10 Oct  3 20:34 usb-Razer_Razer_BlackWidow_V3_Pro_000000000000-if02-event-mouse -> ../event28
lrwxrwxrwx 1 root root   9 Oct  3 20:34 usb-Razer_Razer_BlackWidow_V3_Pro_000000000000-if02-mouse -> ../mouse4
lrwxrwxrwx 1 root root   9 Oct  1 17:28 usb-SteelSeries_Arctis_Pro_Wireless-event-if05 -> ../event7
  ```
</details>

### Packet Captures

<Please see here as we probably need packet captures>
<https://github.com/openrazer/openrazer/wiki/Reverse-Engineering-USB-Protocol>

### Synapse Screenshots
<Screenshots of all available options as seen on Razer Synapse>
(Coming soon)

### PR DESCRIPTION
**Pull request by lissyx**
@z3ntu Rebased against current `master` and looks to be working for me (aside from ripple?)

Fixes #1279

### CODE REVIEW

File: driver/razerkbd_driver.c [hunk 1]
```diff
@@ -2921,19 +2950,20 @@ static ssize_t razer_attr_write_matrix_custom_frame(struct device *dev, struct d
             request.transaction_id.id = 0x1F;
             break;
 
-        case USB_DEVICE_ID_RAZER_BLACKWIDOW_V4_PRO:
-            request = razer_chroma_extended_matrix_set_custom_frame(row_id, start_col, stop_col, (unsigned char*)&buf[offset]);
-            request.transaction_id.id = 0x1F;
-            want_response = false;
-            break;
-
+        case USB_DEVICE_ID_RAZER_BLACKWIDOW_V3_PRO_WIRELESS:
```
Comment (lissyx) [inline driver/razerkbd_driver.c:3044]: Moved here, previous version was badly behaving with:
```
vril 04 07:41:58 portable-alex kernel: ------------[ cut here ]------------
avril 04 07:41:58 portable-alex kernel: WARNING: CPU: 12 PID: 2439186 at /home/alex/codaz/OpenRazer/openrazer/driver/razercommon.c:82 razer_get_usb_response+0x15e/0x1b0 [razerkbd]
avril 04 07:41:58 portable-alex kernel: Modules linked in: razerkbd(O) tls hid_lg_g15 zen_workaround(O) sctp cpuid ch341 usbserial hidp wireguard curve25519_x86_64 libchacha20poly1305 chacha_x86_64 poly1305_x86_64 libcurve25519_generic libchacha ip6_udp_tunnel udp_tunnel rfcomm algif_hash algif_skcipher af_alg ipmi_devintf ipmi_msghandler nf_conntrack_netlink xfrm_user xfrm_algo xt_a>
avril 04 07:41:58 portable-alex kernel:  snd_acp_pci snd_hda_intel snd_usbmidi_lib iwlmvm snd_intel_dspcfg snd_ump uvcvideo btusb snd_pci_acp6x snd_intel_sdw_acpi snd_seq_midi kvm btrtl snd_pci_acp5x videobuf2_vmalloc btbcm snd_seq_midi_event mac80211 uvc snd_hda_codec btintel irqbypass btmtk snd_rn_pci_acp3x videobuf2_memops libarc4 rapl videobuf2_v4l2 snd_rawmidi snd_acp_config snd>
avril 04 07:41:58 portable-alex kernel:  drm_suballoc_helper drm_ttm_helper crct10dif_pclmul ttm crc32_pclmul polyval_clmulni drm_display_helper polyval_generic cec ghash_clmulni_intel rc_core aesni_intel drm_kms_helper ucsi_acpi crypto_simd rtsx_pci_sdmmc xhci_pci mhi_pci_generic nvme r8169 typec_ucsi cryptd psmouse video drm nvme_core xhci_pci_renesas mhi rtsx_pci realtek typec nvm>
avril 04 07:41:58 portable-alex kernel: CPU: 12 PID: 2439186 Comm: openrazer-daemo Tainted: G           O       6.5.0-21-generic #21-Ubuntu
avril 04 07:41:58 portable-alex kernel: Hardware name: LENOVO 21A0CTO1WW/21A0CTO1WW, BIOS R1MET55W (1.25 ) 10/30/2023
avril 04 07:41:58 portable-alex kernel: RIP: 0010:razer_get_usb_response+0x15e/0x1b0 [razerkbd]
avril 04 07:41:58 portable-alex kernel: Code: 00 41 80 7c 24 05 50 77 2c 48 8d 65 d8 5b 41 5c 41 5d 41 5e 41 5f 5d 31 d2 31 c9 31 f6 31 ff 45 31 c0 45 31 c9 e9 92 c1 72 de <0f> 0b c6 42 01 ff e9 c9 fe ff ff 0f b6 1d 78 62 00 00 80 fb 01 0f
avril 04 07:41:58 portable-alex kernel: RSP: 0018:ffffa69bd3977b90 EFLAGS: 00010246
avril 04 07:41:58 portable-alex kernel: RAX: 0000000000000000 RBX: ffffa69bd3977c94 RCX: 0000000000001324
avril 04 07:41:58 portable-alex kernel: RDX: ffffa69bd3977c94 RSI: 0000000000000002 RDI: ffff977af96b8000
avril 04 07:41:58 portable-alex kernel: RBP: ffffa69bd3977bc8 R08: ffffa69bd3977cee R09: 0000000000001324
avril 04 07:41:58 portable-alex kernel: R10: 0000000000000000 R11: 0000000000000000 R12: ffffa69bd3977cee
avril 04 07:41:58 portable-alex kernel: R13: ffff977af96b8000 R14: 0000000000000002 R15: ffffa69bd3977c94
avril 04 07:41:58 portable-alex kernel: FS:  00007f21c3aa16c0(0000) GS:ffff977ed2100000(0000) knlGS:0000000000000000
avril 04 07:41:58 portable-alex kernel: CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
avril 04 07:41:58 portable-alex kernel: CR2: 00007f8532bd9f88 CR3: 0000000136b02000 CR4: 0000000000750ee0
avril 04 07:41:58 portable-alex kernel: PKRU: 55555554
avril 04 07:41:58 portable-alex kernel: Call Trace:
avril 04 07:41:58 portable-alex kernel:  <TASK>
avril 04 07:41:58 portable-alex kernel:  ? show_regs+0x6d/0x80
avril 04 07:41:58 portable-alex kernel:  ? __warn+0x89/0x160
avril 04 07:41:58 portable-alex kernel:  ? razer_get_usb_response+0x15e/0x1b0 [razerkbd]
avril 04 07:41:58 portable-alex kernel:  ? report_bug+0x17e/0x1b0
avril 04 07:41:58 portable-alex kernel:  ? handle_bug+0x51/0xa0
avril 04 07:41:58 portable-alex kernel:  ? exc_invalid_op+0x18/0x80
avril 04 07:41:58 portable-alex kernel:  ? asm_exc_invalid_op+0x1b/0x20
avril 04 07:41:58 portable-alex kernel:  ? razer_get_usb_response+0x15e/0x1b0 [razerkbd]
avril 04 07:41:58 portable-alex kernel:  razer_send_payload.isra.0+0x7f/0x180 [razerkbd]
avril 04 07:41:58 portable-alex kernel:  razer_attr_write_matrix_effect_custom+0x35d/0x5f0 [razerkbd]
avril 04 07:41:58 portable-alex kernel:  dev_attr_store+0x17/0x40
avril 04 07:41:58 portable-alex kernel:  sysfs_kf_write+0x3e/0x60
avril 04 07:41:58 portable-alex kernel:  kernfs_fop_write_iter+0x14f/0x1e0
avril 04 07:41:58 portable-alex kernel:  vfs_write+0x254/0x440
avril 04 07:41:58 portable-alex kernel:  ksys_write+0x73/0x100
avril 04 07:41:58 portable-alex kernel:  __x64_sys_write+0x19/0x30
avril 04 07:41:58 portable-alex kernel:  do_syscall_64+0x5c/0x90
avril 04 07:41:58 portable-alex kernel:  ? srso_alias_return_thunk+0x5/0x7f
avril 04 07:41:58 portable-alex kernel:  ? exit_to_user_mode_prepare+0x30/0xb0
avril 04 07:41:58 portable-alex kernel:  ? srso_alias_return_thunk+0x5/0x7f
avril 04 07:41:58 portable-alex kernel:  ? syscall_exit_to_user_mode+0x37/0x60
avril 04 07:41:58 portable-alex kernel:  ? srso_alias_return_thunk+0x5/0x7f
avril 04 07:41:58 portable-alex kernel:  ? do_syscall_64+0x68/0x90
avril 04 07:41:58 portable-alex kernel:  entry_SYSCALL_64_after_hwframe+0x6e/0xd8
avril 04 07:41:58 portable-alex kernel: RIP: 0033:0x7f21c651b2cf
avril 04 07:41:58 portable-alex kernel: Code: 89 54 24 18 48 89 74 24 10 89 7c 24 08 e8 39 86 f7 ff 48 8b 54 24 18 48 8b 74 24 10 41 89 c0 8b 7c 24 08 b8 01 00 00 00 0f 05 <48> 3d 00 f0 ff ff 77 31 44 89 c7 48 89 44 24 08 e8 8c 86 f7 ff 48
avril 04 07:41:58 portable-alex kernel: RSP: 002b:00007f21c3aa0840 EFLAGS: 00000293 ORIG_RAX: 0000000000000001
avril 04 07:41:58 portable-alex kernel: RAX: ffffffffffffffda RBX: 00000000021beaa0 RCX: 00007f21c651b2cf
avril 04 07:41:58 portable-alex kernel: RDX: 0000000000000001 RSI: 00007f21b4000ba0 RDI: 000000000000000d
avril 04 07:41:58 portable-alex kernel: RBP: 0000000000000001 R08: 0000000000000000 R09: 0000000000000002
avril 04 07:41:58 portable-alex kernel: R10: 00007f21c5bb0900 R11: 0000000000000293 R12: 00007f21c3aa1648
avril 04 07:41:58 portable-alex kernel: R13: 000000000000000d R14: 00007f21b4000ba0 R15: 0000000000000000
avril 04 07:41:58 portable-alex kernel:  </TASK>

File: daemon/openrazer_daemon/hardware/keyboards.py [hunk 2]
```diff
@@ -401,6 +401,23 @@ class RazerBlackWidowV3ProWired(_RippleKeyboard):
...
+    Class for the Razer BlackWidow v3 Pro 2.4 Ghz Wireless
+    """
+
+    USB_PID = 0x025C
+    METHODS = ['get_device_type_keyboard', 'set_wave_effect', 'set_static_effect', 'set_spectrum_effect',
+               'set_reactive_effect', 'set_none_effect', 'set_breath_random_effect', 'set_breath_single_effect', 'set_breath_dual_effect',
+               'set_custom_effect', 'set_key_row', 'get_game_mode', 'set_game_mode', 'get_macro_mode', 'set_macro_mode',
+               'get_macro_effect', 'set_macro_effect', 'get_macros', 'delete_macro', 'add_macro',
+               'set_starlight_random_effect', 'set_starlight_single_effect', 'set_starlight_dual_effect',
+               'set_ripple_effect', 'set_ripple_effect_random_colour',
```
Comment (lissyx) [inline daemon/openrazer_daemon/hardware/keyboards.py]: So as commented in the previous PR, I've added it and:
 - `ripple` does not work on wireless OR when I plug the keyboard
 - same for `reactive` effect, I see nothing happening in both cases

File: driver/razerkbd_driver.c [hunk 3]
```diff
@@ -2921,19 +2950,20 @@ static ssize_t razer_attr_write_matrix_custom_frame(struct device *dev, struct d
             request.transaction_id.id = 0x1F;
             break;
 
-        case USB_DEVICE_ID_RAZER_BLACKWIDOW_V4_PRO:
-            request = razer_chroma_extended_matrix_set_custom_frame(row_id, start_col, stop_col, (unsigned char*)&buf[offset]);
-            request.transaction_id.id = 0x1F;
-            want_response = false;
-            break;
-
```
Comment (z3ntu) [inline driver/razerkbd_driver.c:3019]: Why did you move the code above?

File: driver/razerkbd_driver.c (hunk 1 above)
Comment (z3ntu) [reply driver/razerkbd_driver.c:3044]: It's a warning I added on purpose, if you look into the code at `razercommon.c` line 82. So yes this part looks fine.

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 2 above)
Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/keyboards.py]: So you're saying both ripple and reactive effects also don't work in Wired mode? If it doesn't maybe we should remove it from both Wireless and Wired?

**Comment by z3ntu** [comment]
If you rebase on git master then the CI shouldn't be broken anymore.

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 2 above)
Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: That was at least my experience when I tested the latest PPA version available (3.7.0 I think), on my laptop (ubuntu 23.10). All other effects were doing something, except those two. I am unsure whether it is supposed to be working or not, though.

File: driver/razerkbd_driver.c (hunk 3 above)
Comment (lissyx) [reply driver/razerkbd_driver.c:3019]: it's not removed, just moved below to group `USB_DEVICE_ID_RAZER_BLACKWIDOW_V3_PRO_WIRELESS` with others, I assumed V4 should come after V3 ?

Comment (lissyx) [reply driver/razerkbd_driver.c:3019]: Looks like rebasing on current master there was movement around the V4 one, things should get back in order?

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 2 above)
Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: (also reproduced when using current driver and daemon from git master, with polychromatic-controller, the effect was selectable but nothing would happen and no error reported anywhere as much as I can tell)

File: driver/razerkbd_driver.c [hunk 4]
```diff
@@ -1842,6 +1859,12 @@ static ssize_t razer_attr_write_matrix_effect_reactive(struct device *dev, struc
         request.transaction_id.id = 0x3F;
         break;
 
+    case USB_DEVICE_ID_RAZER_BLACKWIDOW_V3_PRO_WIRELESS:
+        razer_set_device_mode(device, 0x00, 0x00);
```
Comment (z3ntu) [inline driver/razerkbd_driver.c]: This should really not be triggered here. Does the effect really not work when the device is in mode 0x03-0x00 (driver mode)?

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 2 above)
Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/keyboards.py]: I'd say you should add a commit to remove those effects from the Wired version, and not add them in the Wireless version in the first place.

File: driver/razerkbd_driver.c (hunk 4 above)
Comment (lissyx) [reply driver/razerkbd_driver.c]: Judge me if you want, but I got the keyboard wednesday. It wokr with `0x00, 0x00`, I was incorrectly testing (turning the option in polychromatic and waiting, not inputting anything)

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 2 above)
Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py:415]: So, look at the other reply, it seems to work.

File: driver/razerkbd_driver.c (hunk 4 above)
Comment (lissyx) [reply driver/razerkbd_driver.c]: Or I did need `0x03, 0x00` ?

Comment (lissyx) [reply driver/razerkbd_driver.c]: Weird, it was working yesterday now it is not.

Comment (lissyx) [reply driver/razerkbd_driver.c]: Ok so `reactive` works without this call. But Ripple is still not?

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 2 above)
Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py:415]: Or not, I'm a bit lost

Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: Ok, so Ripple works in wired mode, not in wireless for me.

Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: @z3ntu I am unsure if it's a bug driver-side or daemon side, let me know if you have ideas on what to test there?

Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: Could those warnings be related? `razerkbd: Command timed out. status: 04 transaction_id.id: 3f remaining_packets: 00 protocol_type: 00 data_size: 0c, command_class: 0f, command_id.id: 02 Params: 00000800000000000000000000000000 .`

Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/keyboards.py]: Yeah this message is definitely not good. You could try to increase the timeouts in the driver and see if that improves the situation.

Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: What should we do ? Maybe it would be better to just land the current PR without this feature and file a follow up issue ?

Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/keyboards.py]: Removing a broken effect is definitely an option, but I'd still like to see you try my previous comment, so increase timeouts in the driver, maybe that just makes it work.

Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: ok but which timeout? I dont see anything obvious in `razer_attr_write_matrix_effect_custom`

Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: Those?

```
+    case USB_DEVICE_ID_RAZER_BLACKWIDOW_V3_PRO_WIRELESS:
+        *report_index = 0x02;
+        *response_index = 0x02;
+        *wait_min = RAZER_BLACKWIDOW_V3_WIRELESS_WAIT_MIN_US;
+        *wait_max = RAZER_BLACKWIDOW_V3_WIRELESS_WAIT_MAX_US;
+        break;
```

Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/keyboards.py]: yes, wait_min / wait_max there. Update both values to be e.g. 10 times as high as they are right now

Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: Sorry for the delay, I'll try things

Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: Unfortunately, it does not help. I tried hacking a few things, but I dont really have much time to investigate more. Happy to test any suggestion though.

FTR, when I toggle "Ripple" in Polychromatic UI, not only it does not work on the keyboard, but the next effect I select is also not working ; I have to select again a different one before all effects (except Ripple) works.

Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/keyboards.py]: Maybe I've asked this before, but apart from ripple does everything work as expected? We can just get the device in without ripple then. If somebody wants this later they can try and get it working then but we don't need to hold up everything because of ripple.

Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: As much as I can remember, yes, only ripple was behaving erratically.

Comment (lissyx) [reply daemon/openrazer_daemon/hardware/keyboards.py]: Let me rebase and remove ripple for now

**Comment by z3ntu** [comment]
Please check the CI failure, if the device supports the low battery threshold settings, then please add the method to the daemon, if it doesn't support it, then please remove it from the driver to not have a mismatch of supported functionality in the two.

**Comment by lissyx** [comment]
> Please check the CI failure, if the device supports the low battery threshold settings, then please add the method to the daemon, if it doesn't support it, then please remove it from the driver to not have a mismatch of supported functionality in the two.

I'll have a look, thanks for checking

I was wondering why I did not saw that earlier, it's because it was added """recently""" in commit d1231868514c75c736cdad6075a481c40ddea383

**Review by z3ntu — APPROVED** [review]
LGTM, thanks for getting this over the finish line!

**Comment by lissyx** [comment]
I mostly polished what was made ready by others, so thanks @nicfix and @YoMamasNerd
